components/cam/src/physics/cam/micro_p3.F90

```diff
@@ -2814,21 +2815,17 @@ subroutine droplet_activation(t,pres,qv,qc,inv_rho,sup,xxlv,npccn,log_predictNc,
    if (log_predictNc) then
       ! for predicted Nc, use activation predicted by aerosol scheme
       ! note that this is also applied at the first time step
-      if (sup.gt.1.e-6) then
+      if (qc .gt. qsmall) then
+! +++ JShpund: 1. Here we need the same condition for nucleation performs at the beginning of the scheme.
+!                 The condition is either on mass "qsmall", liquid supersaturation "sup" or both. (?).
+!                 Given saturation adjustment is fundemental to macrophysics, which values are expected here?
+!              2. "npcc" is in-cloud value here.
+!              3. No need to add "npccn" to the tendency (compared to MG2)!
+!              3. The corresponding mass tendency for nucleation is done in macrophysics and thus is erased here
+!              *Erase When Done*
          ncnuc = npccn
-         !TODO Limit qcnuc so that conditions never become sub-saturated
-         qcnuc = ncnuc*cons7
+         qcnuc = 0.0_rtype
```

Comment:
I don't really understand the confusion here. I agree that all that should be done here is just rename npccn => ncnuc and quit. Given that this function has become so trivial, I'd suggest:

1. rename npccn => ncnuc in the call signature of p3_main. Make it an INOUT variable. 
2. delete subroutine droplet_activation and its call. Now the existing wiring for ncnuc should make sure ncnuc is added to the net microphysical tendency even though it is just being passed in from an external calculation (it's worth checking this actually happens). 
3. Delete sup - we shouldn't need it anywhere in our code. 

Does this make sense? I can never keep clear whether npccn is an in-cloud or a cell-average value, but we should make sure we are using it consistently.

Reply:
@PeterCaldwell @AaronDonahue 

OK, makes sense. 
This leave us with the erroneous position of the following initialization part which is being done for in-cloud fields:
```
! initialize microphysics processes tendency output
p3_tend_out(i,k,42) = qc(i,k)    ! Liq. microphysics tendency, initialize
p3_tend_out(i,k,43) = nc(i,k)    ! Liq. # microphysics tendency, initialize
p3_tend_out(i,k,44) = qr(i,k)    ! Rain microphysics tendency, initialize
p3_tend_out(i,k,45) = nr(i,k)    ! Rain # microphysics tendency, initialize
p3_tend_out(i,k,46) = qitot(i,k) ! Ice  microphysics tendency, initialize
p3_tend_out(i,k,47) = nitot(i,k) ! Ice  # microphysics tendency, initialize
p3_tend_out(i,k,48) = qv(i,k)    ! Vapor  microphysics tendency, initialize
p3_tend_out(i,k,49) = th(i,k)    ! Pot. Temp. microphysics tendency, initialize
```

Reply:
I don't understand the confusion. If npccn is a cell-average and p3_tend_out(i,k,43) is supposed to be an in-cloud value, set p3_tend_out(i,k,43)=npccn/cloud_fraction. Or if p3_tend_out(i,k,43) is supposed to be a cell ave and npccn is an in-cloud value, set p3_tend_out(i,k,43)=npccn*cloud_fraction. What am I missing here?

Reply:
We are on the same page as for the nucleation correction. 
However, for example the tendency for `nc` around line number 1145 in `micro_p3.F90` looks like:
`p3_tend_out(i,k,43) = ( nc(i,k)    - p3_tend_out(i,k,43) ) * odt ! Liq. # microphysics tendency, measure`
The RHS is assumes to have cell-average values, but the `p3_tend_out(i,k,43) `of the RHS is being initialize with in-cloud values. 

Didn't want to change that position without your OK.

Reply:
Ah - I see now. Thanks for pointing this out. I hate it when "tendencies" are used as temporary holding space for state variables - so confusing!!! 

There are 2 possible fixes: 

1. leave things as they are but add npccn to p3_tend_out(:,:,43) after its final value has been calculated
2. move the initialization of p3_tend_out(:,:,43) to before p3_main_pre_loop.

My preference is actually a variant on option 2: I'd suggest deleting all lines around L795 which initialize tendencies with state values and replace them with lines like qc_old(:,:) = qc(:,:) around L741 (just after the get_latent_heat call). Then use these "_old" variables instead of the states masquerading as functions for the actual tendency calculations. Does that make sense? It increases our memory usage a bit, but makes code much more clear.